samza-rest/src/main/java/org/apache/samza/rest/model/ApplicationInfo.java

```diff
@@ -0,0 +1,295 @@
+  private String _applicationTags;
+
+  @JsonProperty("finishedTime")
+  private String _finishedTime;
+
+  @JsonProperty("preemptedResourceMB")
+  private String _preemptedResourceMB;
+
+  @JsonProperty("preemptedResourceVCores")
+  private String _preemptedResourceVCores;
+
+  @JsonProperty("numNonAMContainerPreempted")
+  private String _numNonAMContainerPreempted;
+
+  @JsonProperty("numAMContainerPreempted")
+  private String _numAMContainerPreempted;
+
+  @JsonProperty("progress")
+  private int _progress;
+
+  @JsonProperty("name")
+  private String _name;
+
+  @JsonProperty("startedTime")
+  private long _startedTime;
+
+  @JsonProperty("elapsedTime")
+  private long _elapsedTime;
+
+  @JsonProperty("diagnostics")
+  private String _diagnostics;
+
+  @JsonProperty("trackingUrl")
+  private String _trackingUrl;
+
+  @JsonProperty("queue")
+  private String _queue;
+
+  @JsonProperty("allocatedMB")
+  private int _allocatedMB;
+
+  @JsonProperty("allocatedVCores")
+  private int _allocatedVCores;
+
+  @JsonProperty("runningContainers")
+  private int _runningContainers;
+
+  @JsonProperty("memorySeconds")
+  private int _memorySeconds;
+
+  @JsonProperty("vcoreSeconds")
+  private int _vcoreSeconds;
+
+  ApplicationInfo() {
+
+  }
+
+  public ApplicationInfo(long finishTime, String amContainerLogs, String trackingHistory, String state, String user,
+      String id, String clusterId, String finalStatus, String amHostHttpAddress, String applicationType,
+      String applicationTags, String finishedTime, String preemptedResourceMB, String preemptedResourceVCores,
+      String numNonAMContainerPreempted, String numAMContainerPreempted, int progress, String name, long startedTime,
+      long elapsedTime, String diagnostics, String trackingUrl, String queue, int allocatedMB, int allocatedVCores,
+      int runningContainers, int memorySeconds, int vcoreSeconds) {
+    _finishTime = finishTime;
+    _amContainerLogs = amContainerLogs;
+    _trackingHistory = trackingHistory;
+    _state = state;
+    _user = user;
+    _id = id;
+    _clusterId = clusterId;
+    _finalStatus = finalStatus;
+    _amHostHttpAddress = amHostHttpAddress;
+    _applicationType = applicationType;
+    _applicationTags = applicationTags;
+    _finishedTime = finishedTime;
+    _preemptedResourceMB = preemptedResourceMB;
+    _preemptedResourceVCores = preemptedResourceVCores;
+    _numNonAMContainerPreempted = numNonAMContainerPreempted;
+    _numAMContainerPreempted = numAMContainerPreempted;
+    _progress = progress;
+    _name = name;
+    _startedTime = startedTime;
+    _elapsedTime = elapsedTime;
+    _diagnostics = diagnostics;
+    _trackingUrl = trackingUrl;
+    _queue = queue;
+    _allocatedMB = allocatedMB;
+    _allocatedVCores = allocatedVCores;
+    _runningContainers = runningContainers;
+    _memorySeconds = memorySeconds;
+    _vcoreSeconds = vcoreSeconds;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Let's not override hashcode and equals. There's no particular reason to expect content based equality for this class.